docs: Add to FAQ: When / how to use Linked Ingredients (#4897)
mealie-recipes/mealie: docs/docs/documentation/getting-started/faq.md

    4. Click 'Update'

??? question "Why Link Ingredients to a Recipe Step?"

>     **Why Link Ingredients to a Recipe Step?**
> 
>     Mealie allows you to link ingredients to specific steps in a recipe, ensuring you know exactly when to add each ingredient during the cooking process.
> 
>     #### Link Ingredients to Steps in a Recipe

Reviewer: Ah i think my comment was a bit confusing. I did not mean the first title within the card that repeasts the question (this one should be in the table of contents) but every other title within a single faq answer should not be an title to not clutter the table of content. The desired result is to just have section titles and the questions in the TOC

```suggestion
    ### Why Link Ingredients to a Recipe Step?

    Mealie allows you to link ingredients to specific steps in a recipe, ensuring you know exactly when to add each ingredient during the cooking process.

    **Link Ingredients to Steps in a Recipe**
    
```

    4. Click 'Update'

> ??? question "When would I use Linked Ingredients in a Recipe?"
> 
>     ### When would I use Linked Ingredients in a Recipe?

Reviewer: ```suggestion
??? question "Why Link Ingredients to a Recipe Step?"

    ### Why Link Ingredients to a Recipe Step??
```

    Mealie allows you to link ingredients to specific steps in a recipe, ensuring you know exactly when to add each ingredient during the cooking process.

    #### Link Ingredients to Steps in a Recipe
    1. Go to a recipe
    2. Click the Edit button/icon
    3. Scroll down to the step you want to link ingredients to
    4. Click the ellipsis button next to the step and click 'Link Ingredients'
    5. Check off the Ingredient(s) that you want to link to that step
    6. Optionally, click 'Next step' to continue linking remaining ingredients to steps, or click 'Save' to Finish
    7. Click 'Save' on the Recipe

    You can optionally link the same ingredient to multiple steps, which is useful for prepping an ingredient in one step and using it in another.

> ??? question "Should I setup Email / What is Email used for?"

Reviewer: I would probably move this to "Technical Consideration" and rename it to "Why setup email?"

Also we need a note in here that these links can be distrirbuted by other means as well.

## Security and Maintenance

??? question "How can I use Mealie externally?"

    ### How can I use Mealie externally?

    By default, Mealie is accessible only on the local network where it is set up. Certain features, such as Shopping Lists, are helpful to access when you are not on your local network. If you want to access Mealie externally, there are several methods with examples listed below, each with its own security considerations. Choose the option that best suits your network's needs. Since every network is unique, we cannot provide specific steps for this process and encourage you to seek further assistance on our Discord.

    1. Reverse Proxy Tunnel (e.g., Nginx, Cloudflare Zero Tunnel Network Access)
    2. VPN
    3. Port Forwarding (Note: Exposing internal devices directly to the internet poses a security risk. This method is not recommended without additional safeguards or security measures in place.)

> 

Reviewer 2: I'm not sure I'm comfortable with publishing this as is at this time. I'd rather remove it altogether and merge the PR with the rest.

The reason for this is that there's no way to give a definitive statement that is applicable to everyone, given how docker works..
If someone's computer is exposed, Mealie can be exposed.
We can't really control/influence any of that.

Reviewer 2: And, for someone who doesn't understand those nuances.. they definitely shouldn't be exposing anything.

Author: There is a community guide on setting up a reverse proxy. We could instead link to that and say "Proceed with caution as any method to expose Mealie to the internet could put your home network at risk" or something like that?
